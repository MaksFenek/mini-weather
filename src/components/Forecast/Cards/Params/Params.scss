.params {
  ul {
    list-style: none;
    li {
      display: flex;
      align-items: center;
      font-size: 20px;
      margin-bottom: 24px;
      span {
        margin-left: 12px;
      }
    }
  }
}
