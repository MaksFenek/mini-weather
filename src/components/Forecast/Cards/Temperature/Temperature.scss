.temperature {
  margin-top: 24px;
  &-value {
    display: flex;

    h2 {
      margin-left: 24px;
      font-size: 2.5rem;
    }
  }

  h3 {
    font-size: 2rem;
    font-weight: normal;
    letter-spacing: 0.7px;
    margin: 12px 0;
  }
}
