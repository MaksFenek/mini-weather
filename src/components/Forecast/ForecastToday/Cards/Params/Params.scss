.params {
  ul {
    list-style: none;
    li {
      display: flex;
      align-items: center;
      font-size: 20px;
      margin-bottom: 24px;
      p {
        margin-left: 12px;
        span {
          font-weight: bold;
        }
      }
    }
  }
}
