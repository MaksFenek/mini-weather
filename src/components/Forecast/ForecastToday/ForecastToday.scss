.forecast {
  grid-area: forecast;
  background: #bbdefb;
  border-radius: 20px;
  margin-top: 14px;
  padding: 24px;
  &-header {
    font-size: 1.8rem;
    font-weight: bold;
    display: flex;
    justify-content: space-between;

    .date {
      font-weight: normal;
      font-size: 1.2rem;
    }
  }

  &-content {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }
}
