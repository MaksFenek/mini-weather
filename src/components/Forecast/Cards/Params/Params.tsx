import React from 'react';
import './Params.scss';

interface IParams {
  humidity: number | '--';
  pressure: number | '--';
  wind: {
    speed: number;
    deg: number;
  };
}

const Params = ({ humidity = '--', pressure = '--', wind }: IParams) => {
  return (
    <div className='forecast-content__params'>
      <div className='params'>
        <ul>
          <li>
            <svg className='icon' height='36px' width='36px'>
              <use xlinkHref={`/src/Icons/params.svg#humidity`} />
            </svg>
            <span>Humidity: {humidity} %</span>
          </li>
          <li>
            <svg className='icon' height='36px' width='36px'>
              <use xlinkHref={`/src/Icons/params.svg#barometer`} />
            </svg>
            <span>Pressure: {pressure} hPa</span>
          </li>
          <li>
            <svg
              className='icon'
              height='52px'
              width='52px'
              style={{ margin: '-8px' }}
            >
              <use
                xlinkHref={`/src/Icons/compass.svg#main
              `}
              ></use>
              <use
                style={{
                  transform: `rotate(${wind.deg + 90}deg)`,
                  transformOrigin: 'center',
                }}
                xlinkHref={`/src/Icons/compass.svg#arrow`}
              ></use>
            </svg>
            <span>Wind: {wind ? wind.speed : '--'} mph</span>
          </li>
        </ul>
      </div>
    </div>
  );
};

export default Params;
