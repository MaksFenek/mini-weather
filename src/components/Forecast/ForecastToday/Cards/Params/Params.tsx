// React
import React from 'react';

// Styles
import './Params.scss';

// Types
interface IParams {
  humidity: number | '--';
  pressure: number | '--';
  wind: {
    speed: number;
    deg: number;
  };
}

const Params: React.FC<IParams> = ({
  humidity = '--',
  pressure = '--',
  wind,
}) => {
  return (
    <div className='params'>
      <ul>
        <li>
          <svg className='icon' height='36px' width='36px'>
            <use xlinkHref={`icons/params.svg#humidity`} />
          </svg>
          <p>
            Humidity: <span>{humidity}</span> %
          </p>
        </li>
        <li>
          <svg className='icon' height='36px' width='36px'>
            <use xlinkHref={`icons/params.svg#barometer`} />
          </svg>
          <p>
            Pressure: <span>{pressure}</span> hPa
          </p>
        </li>
        <li>
          <svg
            className='icon'
            height='52px'
            width='52px'
            style={{ margin: '-7px' }}
          >
            <use
              xlinkHref={`icons/compass.svg#main
              `}
            ></use>
            <use
              style={{
                transform: `rotate(${wind.deg + 270}deg)`,
                transformOrigin: 'center',
              }}
              xlinkHref={`icons/compass.svg#arrow`}
            ></use>
          </svg>
          <p>
            Wind: <span>{wind ? wind.speed : '--'}</span> mph
          </p>
        </li>
      </ul>
    </div>
  );
};

export default Params;
