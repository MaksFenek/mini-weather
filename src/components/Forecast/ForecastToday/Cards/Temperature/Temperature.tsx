// React
import React from 'react';

// Styles
import './Temperature.scss';

// Types
interface ITemperature {
  temp: number | '--';
  description: string;
}

const Temperature: React.FC<ITemperature> = ({ temp, description }) => {
  return (
    <div className='temperature'>
      <div className='temperature-value'>
        <svg className='icon' height='96px' width='96px'>
          <use
            aria-label='icon'
            xlinkHref={`icons/icons.svg#${description.toLowerCase()}`}
          />
        </svg>
        <h2 aria-label='temp'>{temp} Â°C</h2>
      </div>
      <h3 aria-label='temp'>{description}</h3>
    </div>
  );
};

export default Temperature;
