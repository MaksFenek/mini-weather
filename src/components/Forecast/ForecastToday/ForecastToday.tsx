// React
import React, { ReactNode } from 'react';

// Styles
import './ForecastToday.scss';

// Types
import { IDate } from '../../Main/Main';
interface IForecastToday {
  city: string;
  children: ReactNode;
  date: IDate | '--';
}

const ForecastToday: React.FC<IForecastToday> = ({ city, children, date }) => {
  return (
    <div className='forecast' aria-label='forecast'>
      <div className='forecast-header'>
        <span className='city' aria-label='city'>
          {city}
        </span>
        {date !== '--' && (
          <span
            className='date'
            aria-label='date'
          >{`${date.day} ${date.month}, ${date.name}`}</span>
        )}
      </div>
      <div className='forecast-content' aria-label='children'>
        {children}
      </div>
    </div>
  );
};

export default ForecastToday;
