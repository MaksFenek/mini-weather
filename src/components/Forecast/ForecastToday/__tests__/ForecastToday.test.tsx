import React from 'react';
import ForecastToday from '../ForecastToday';
import { mount } from 'enzyme';
import Temperature from '../Cards/Temperature/Temperature';
import Params from '../Cards/Params/Params';
import renderer from 'react-test-renderer';

describe('<ForecastToday/>', () => {
  const city = 'Toronto';
  const date = {
    name: 'Monday',
    day: 13,
    month: 'January',
  };
  const temp = 23;
  const description = 'Clouds';
  const humidity = 40;
  const pressure = 1000;
  const wind = {
    speed: 2.3,
    deg: 120,
  };

  const wrapper = mount(
    <ForecastToday city={city} date={date}>
      <Temperature temp={temp} description={description} />
      <Params humidity={humidity} pressure={pressure} wind={wind} />
    </ForecastToday>
  );
  const tree = renderer
    .create(
      <ForecastToday city={city} date={date}>
        <Temperature temp={temp} description={description} />
        <Params humidity={humidity} pressure={pressure} wind={wind} />
      </ForecastToday>
    )
    .toJSON();
  it('should render correctly', () => {
    expect(wrapper.find(`[aria-label='city']`).text()).toEqual(city);
    expect(wrapper.find(`[aria-label='date']`).text()).toEqual(
      `${date.day} ${date.month}, ${date.name}`
    );
    expect(wrapper.find(`[aria-label='children']`).children()).toHaveLength(2);
    expect(wrapper).toMatchSnapshot();
  });
});
