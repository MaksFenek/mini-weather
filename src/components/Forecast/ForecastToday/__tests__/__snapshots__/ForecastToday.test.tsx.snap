// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastToday/> should render correctly 1`] = `ReactWrapper {}`;
